server-spi-private/src/main/java/org/keycloak/cluster/ClusterProvider.java

```diff
@@ -78,7 +80,11 @@ public interface ClusterProvider extends Provider {
      * @param ignoreSender if true, then sender node itself won't receive the notification
      * @param dcNotify Specify which DCs to notify. See {@link DCNotify} enum values for more info
      */
-    void notify(String taskKey, ClusterEvent event, boolean ignoreSender, DCNotify dcNotify);
+    default void notify(String taskKey, ClusterEvent event, boolean ignoreSender, DCNotify dcNotify) {
+        notify(taskKey, Collections.singleton(event), ignoreSender, dcNotify);
+    }
+
+    void notify(String taskKey, Collection<? extends ClusterEvent> events, boolean ignoreSender, DCNotify dcNotify);
```

Comment:
I know this is a private SPI, still let's try to be backwards compatible, so people can use their provider. As you mentioned you want to keep both methods, as it is more convenient to use the one with once event if there not a collection. 

So as discussed on the call, please do the following: 
* The current default implementation would move down to the InfinspanClusterProvider
* A new default implementation for the new method which would call the old one
* A matching JavaDoc for the new method
* An upgrading guide entry mentioning the new method that people should implement it
* A new follow-up issue for Keycloak 27 to remove the default method where we remove the default method, so people will change their providers for a more efficient method.

Reply:
follow up: https://github.com/keycloak/keycloak/issues/30457